catalog/core/catalog-core-solrcommands/src/test/java/org/codice/ddf/commands/solr/TestBackupCommand.java

```diff
@@ -142,6 +199,296 @@ public void testSystemPropertiesNotSet() throws Exception {
+
+        // Perform Test
+        backupCommand1.doExecute();
+
+        // Verify
+        assertThat(consoleOutput.getOutput(),
+                containsString(String.format(
+                        "Backing up collection [%s] to shared location [%s] using backup name [%s_",
+                        DEFAULT_CORE_NAME,
+                        backupCommand1.backupLocation,
+                        DEFAULT_CORE_NAME)));
+        assertThat(consoleOutput.getOutput(), containsString("Solr Cloud backup request Id:"));
+
+        String requestId = getRequestId(consoleOutput.getOutput());
+        consoleOutput.reset();
+
+        BackupCommand backupCommand2 = new BackupCommand() {
+            @Override
+            public SolrClient getSolrClient() {
+                return miniSolrCloud.getSolrClient();
+            }
+
+            // We get the solr client from the MiniSolrCloudCluster, so we don't
+            // want to shut it down after each test. We don't create a MiniSolrCloudCluster
+            // for each test to reduce the time it takes to run the tests.
+            @Override
+            void shutdown(SolrClient client) {}
+        };
+        backupCommand2.asyncBackupStatus = true;
+        backupCommand2.asyncBackupReqId = requestId;
+
+        backupCommand2.doExecute();
+        String status = getRequestStatus(consoleOutput.getOutput());
+
+        while(StringUtils.equals(status, RequestStatusState.RUNNING.getKey())) {
+            TimeUnit.SECONDS.sleep(1);
+            consoleOutput.reset();
+            backupCommand2.doExecute();
+            status = getRequestStatus(consoleOutput.getOutput());
+        }
+
+        assertThat(status, is(RequestStatusState.COMPLETED.getKey()));
+    }
+
+    @Test
+    public void testGetSolrCloudAsynchronousBackupStatusNoRequestId() throws Exception {
+
+        // Set the solr client type system property so that the
+        // BackupCommand knows that it needs to backup solr cloud.
+        setupSolrClientType(SOLR_CLOUD_CLIENT_TYPE);
+
+        // Setup BackupCommand
+        BackupCommand backupCommand = new BackupCommand() {
+            @Override
+            SolrClient getSolrClient() {
+                return miniSolrCloud.getSolrClient();
+            }
+
+            // We get the solr client from the MiniSolrCloudCluster, so we don't
+            // want to shut it down after each test. We don't create a MiniSolrCloudCluster
+            // for each test to reduce the time it takes to run the tests.
+            @Override
+            void shutdown(SolrClient client) {}
+        };
+        backupCommand.asyncBackupStatus = true;
+
+        backupCommand.doExecute();
+
+        assertThat(consoleOutput.getOutput(),
+                containsString("Insufficient options. Run solr:backup --help for usage details."));
+    }
+
+    @Test
+    public void testGetSolrCloudAsynchronousBackupStatusNoStatusOption() throws Exception {
```

Comment:
Isn't this more a test of the SolrCloudBackup without any options rather than asynchronous backup?